src/rect.rs

```diff
@@ -348,6 +348,16 @@ where
     }
 }
 
+impl<T, U> TypedRect<T, U>
+where
+    T: Copy + One + Add<Output = T> + Div<Output = T>,
+{
+    pub fn center(&self) -> TypedPoint2D<T, U> {
+        let two = T::one() + T::one();
+        ((self.origin.to_vector() + self.origin.to_vector() + self.size.to_vector()) / two).to_point()
```

Comment:
with this implementation, it doesn't look like we need to add the origin twice before dividing it

Reply:
true, I guess we could do origin + size/2